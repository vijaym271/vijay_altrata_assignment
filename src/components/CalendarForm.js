import { useState } from "react";
import { useDispatch } from "react-redux";
import { addCalender } from "../redux/slices/calendarSlice";

const CalendarForm = () => {
    const dispatch = useDispatch();
    const [date, setDate] = useState('')

    const handleSubmit = (e) => {
        e.preventDefault();
        const dateRegEx = /\d{1,2}\/\d{1,2}\/\d{2,4}/;
        const [DD, MM] = date.split('/');
        const isValid = (DD <= 31 && MM <= 12)
        if (date.trim().length > 0 && dateRegEx.test(date.trim()) && isValid) {
            dispatch(addCalender(date))
            setDate('')
        }
    }

    const handleChange = (e) => setDate(e.target.value);

    return <div className="col-md-3 mx-auto">
        <form data-testid="calendarForm" onSubmit={handleSubmit}>
            <div className="row">
                <div className="col-md-7">
                    <input data-testid="calendarInput" className="form-control" name="calendar" value={date} onChange={handleChange} placeholder="eg - 20/5/2023" />
                </div>
                <div className="col-md-5">
                    <button data-testid="calendarFormSubmit" className="btn btn-primary" type="submit">Get Calendar</button>
                </div>
            </div>
        </form>
    </div>
}
export default CalendarForm;