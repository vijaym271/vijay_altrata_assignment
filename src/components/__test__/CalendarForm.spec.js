import { render, screen, fireEvent } from '@testing-library/react'
import CalendarForm from '../CalendarForm';
import renderer from 'react-test-renderer'
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureMockStore();

describe("CalendarForm", () => {
    let store;
    beforeEach(() => {
        store = mockStore(["3/10/2022"]);
    });
    it("render form fields", () => {
        render(
            <Provider store={store}>
                <CalendarForm />
            </Provider>
        );
        const calendarInput = screen.getByTestId("calendarInput");
        const calendarFormSubmit = screen.getByTestId("calendarFormSubmit");
        expect(calendarInput).toBeInTheDocument();
        expect(calendarFormSubmit).toBeInTheDocument();
    })

    it('submit calendarForm', () => {
        const mockSubmit = jest.fn();
        render(<Provider store={store}>
            <CalendarForm />
        </Provider>);
        const calendarForm = screen.getByTestId("calendarForm");
        const calendarInput = screen.getByTestId("calendarInput");
        fireEvent.change(calendarInput, { target: { value: '27/01/2023' } });
        calendarForm.addEventListener('submit', mockSubmit);
        fireEvent.submit(calendarForm)
        expect(mockSubmit).toHaveBeenCalledTimes(1);
        expect(mockSubmit).toHaveBeenCalledWith(expect.any(Object));
    })
})

describe("CalendarForm Snapshot", () => {
    let store;
    beforeEach(() => {
        store = mockStore(["3/10/2022"]);
    });
    it('should match dom snapshot', () => {
        const tree = renderer.create(
            <Provider store={store}>
                <CalendarForm />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})

