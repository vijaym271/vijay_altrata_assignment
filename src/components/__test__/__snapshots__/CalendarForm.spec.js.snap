// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Form Snapshot should match dom snapshot 1`] = `
<div
  className="col-md-3 mx-auto"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="row"
    >
      <div
        className="col-md-7"
      >
        <input
          className="form-control"
          data-testid="calendarInput"
          name="calendar"
          onChange={[Function]}
          placeholder="eg - 20/5/2023"
          value=""
        />
      </div>
      <div
        className="col-md-5"
      >
        <button
          className="btn btn-primary"
          data-testid="calendarFormSubmit"
          type="submit"
        >
          Get Calendar
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`CalendarForm Snapshot should match dom snapshot 1`] = `
<div
  className="col-md-3 mx-auto"
>
  <form
    data-testid="calendarForm"
    onSubmit={[Function]}
  >
    <div
      className="row"
    >
      <div
        className="col-md-7"
      >
        <input
          className="form-control"
          data-testid="calendarInput"
          name="calendar"
          onChange={[Function]}
          placeholder="eg - 20/5/2023"
          value=""
        />
      </div>
      <div
        className="col-md-5"
      >
        <button
          className="btn btn-primary"
          data-testid="calendarFormSubmit"
          type="submit"
        >
          Get Calendar
        </button>
      </div>
    </div>
  </form>
</div>
`;
